### PR TITLE
Tutorial/vertex cover

### PR DESCRIPTION
**Pull request by ahao27**
## Changes
1. Added a tutorial notebook file, "vertex_cover.ipynb," under the `docs/tutorial/`.
2. Fixed the index.md file.
3. Updated the _toc.yml file.

## Description
This tutorial demonstrates how to solve the minimum Vertex Cover problem using Qamoile and Jijmodleing.

## Issue
Currently, in the last cell of "vertex_cover.ipynb," an error occurs when building with `Jupyter-book`:
` NameError: name 'cbclib' is not defined.`

### CODE REVIEW

**Review by Jacomichi — CHANGES_REQUESTED** [review]
Please fix the following points.
- Please change the title to “QAOA for the Vertex Cover”.
- Please raise the upper limit so that `minimize` will work until convergence.
- Please delete “Comparing with the Exact Solution” as it is not included in the other tutorials.
- Please remove the output of `qaoa_counts.get_counts()`.

**Comment by ahao27** [comment]
## Change
- [x] Please change the title to “QAOA for the Vertex Cover”.
- [x] Please raise the upper limit so that minimize will work until convergence.
=> The minimize now shows succes in nfev=1071.
- [x] Please delete “Comparing with the Exact Solution” as it is not included in the other tutorials.
- [x] Please remove the output of qaoa_counts.get_counts().

File: docs/tutorial/vertex_cover.ipynb [hunk 1]
```diff
@@ -0,0 +1,672 @@
...
+    "id": "GuAxhE9ulPV8"
+   },
+   "source": [
+    "## Converting Compiled Instance to QAOA Circuit and Hamiltonian\n",
+    "\n",
+    "We generate the QAOA circuit and Hamiltonian from the compiled Instance. The converter used to generate these is `qm.qaoa.QAOAConverter`.\n",
+    "\n",
+    "By creating an instance of this class and using `ising_encode`, you can internally generate the Ising Hamiltonian from the compiled Instance. Parameters that arise during the conversion to QUBO can also be set here. If not set, default values are used.\n",
+    "\n",
+    "Once the Ising Hamiltonian is generated, you can generate the QAOA quantum circuit and the Hamiltonian respectively. These can be executed using the `get_qaoa_ansatz` and `get_cost_hamiltonian` methods. The number of QAOA layers, $P$, is fixed to be $4$ here.   "
```
Comment (Jacomichi) [inline docs/tutorial/vertex_cover.ipynb:276]: $P$ -> $p$

File: notebook/Tutorial_VertexCover.ipynb [hunk 2]
```diff
@@ -0,0 +1,897 @@
...
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "sampleset = qaoa_converter.decode(qk_transpiler, job_result.data[\"meas\"])\n",
```
Comment (Jacomichi) [inline notebook/Tutorial_VertexCover.ipynb:721]: Can you change the plot code like quickstart (https://jij-inc.github.io/Qamomile/quickstart.html#decode-the-result)
The reason is in https://github.com/Jij-Inc/Qamomile/issues/76#issue-2562951830

**Review by Jacomichi — CHANGES_REQUESTED** [review]
I have left the tiny comments

File: docs/tutorial/vertex_cover.ipynb [hunk 3]
```diff
@@ -0,0 +1,674 @@
...
+    "nx.draw(G, node_color=color, pos=positions, with_labels=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "fulDayMynzrH"
+   },
+   "source": [
+    "## Formulation of the Vertex Cover Problem\n",
```
Comment (Jacomichi) [inline docs/tutorial/vertex_cover.ipynb:99]: Tex in this section is broken in local build

File: docs/tutorial/vertex_cover.ipynb [hunk 4]
```diff
@@ -0,0 +1,674 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "fmNHeUw7lXVI"
+   },
+   "source": [
+    "## Converting the Obtained QAOA Circuit and Hamiltonian for Qiskit\n",
+    "\n",
+    "We generate the QAOA circuit and Hamiltonian from the compiled instance using the `qamomile.qiskit.QiskitTranspiler` and `qamomile.qiskit.transpile_hamiltonian` converters. By utilizing these two methods, we can transform the QAOA circuit and Hamiltonian into a format compatible with Qiskit. This allows us to leverage Qiskit's quantum computing framework to execute and analyze."
```
Comment (Jacomichi) [inline docs/tutorial/vertex_cover.ipynb:334]: `transpile_hamiltonian` is the method of `QiskitTranspiler`, so this explanation is not correct.

**Review by Jacomichi — CHANGES_REQUESTED** [review]
I have left the comments, please check

File: docs/tutorial/vertex_cover.ipynb (hunk 3 above)
Comment (ahao27) [reply docs/tutorial/vertex_cover.ipynb:99]: Fixed by spliting the cell to several.

File: docs/tutorial/vertex_cover.ipynb (hunk 4 above)
Comment (ahao27) [reply docs/tutorial/vertex_cover.ipynb]: Fixed!

**Review by Jacomichi — APPROVED** [review]
LGTM

**Pull request merged**